traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -650,10 +673,15 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 		if userErr != nil || sysErr != nil {
 			return nil, 0, errors.New("Forbidden"), sysErr, http.StatusForbidden, nil
 		}
+
+		var joinSubQuery string
+		if version.Major >= 3 {
```

Comment:
should check that the version is non-null before dereferencing property to avoid segfault

also, though, I don't know why we decided `APIInfo.Version` should be a pointer value. Maybe for plugins? If I were you, I'd check it in the caller and make this function take just a regular `api.Version`. Hopefully at some point we can put that check into `api.NewInfo` and make it not give out possibly null API versions. Since if you can't tell what version someone requested, you already know they requested an invalid URI, so you can just error out right there.

Reply:
Added null checks to calling functions and changed the function signature to accept a non-pointer version in 50ae3819eb.